Gems/Atom/RHI/Code/Include/Atom/RHI/CpuProfilerImpl.h

```diff
@@ -134,6 +139,12 @@ namespace AZ
             AZStd::shared_mutex m_shutdownMutex;
 
             bool m_initialized = false;
+
+            bool m_continuousCaptureInProgress = false;
+
+            // Stores multiple frames of profiling data, size is controlled by MaxFramesToSave. Flushed when EndContinuousCapture is called.
+            // Deque so that we can have fast append of new data + removal of old profiling data.
+            AZStd::deque<TimeRegionMap> m_continuousCaptureData;
```

Comment:
Question: thoughts about using deque here over a vector? I chose it so that we could have O(1) removal of old profiling data if we are at `MaxFramesToSave` but it may perform worse than just a plain old vector due to cache effects.

Reply:
You could use a circular or ring buffer to get the benefits of both O(1) removal/insertion in a linear range of memory like a vector.

Reply:
That's exactly what I was looking for, thanks! Moved to `AZStd::ring_buffer`